src/gui/processing/models/qgsmodelcomponentgraphicitem.cpp

```diff
@@ -563,15 +567,20 @@ void QgsModelComponentGraphicItem::fold( Qt::Edge edge, bool folded )
 
   // TODO - this is not so nice, consider moving this to model class
   if ( QgsProcessingModelChildAlgorithm *child = dynamic_cast<QgsProcessingModelChildAlgorithm *>( mComponent.get() ) )
+  {
     mModel->childAlgorithm( child->childId() ).setLinksCollapsed( edge, folded );
+  }
   else if ( QgsProcessingModelParameter *param = dynamic_cast<QgsProcessingModelParameter *>( mComponent.get() ) )
-    mModel->parameterComponent( param->parameterName() ).setLinksCollapsed( edge, folded );
+  {
+    QgsProcessingModelParameter parameterComponent = mModel->parameterComponent( param->parameterName() );
+    parameterComponent.setLinksCollapsed( edge, folded );
```

Comment:
This is broken -- you are setting setLinksCollapsed on a copy which you do nothing with. The original code worked because parameterComponent returns a reference.